mavros/CMakeLists.txt

```diff
@@ -38,6 +38,45 @@ endif()
 include(EnableCXX11)
 include(MavrosMavlink)
 
+## Find GeographicLib
+list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
+find_package(GeographicLib 1.40 COMPONENTS SHARED)
+if(NOT GeographicLib_FOUND)
+  include(ExternalProject)
+
+  # The configure step fails at catkin_package() if this directory doesn't exist yet
+  file(MAKE_DIRECTORY ${CATKIN_DEVEL_PREFIX}/include)
+
+  ExternalProject_Add(
+      GeographicLib
+      GIT_REPOSITORY git://git.code.sourceforge.net/p/geographiclib/code
+      PATCH_COMMAND mkdir -p ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}/data
+      CONFIGURE_COMMAND cd ${CMAKE_BINARY_DIR}/GeographicLib-prefix/src/GeographicLib && cmake -D CMAKE_INSTALL_PREFIX=${CATKIN_DEVEL_PREFIX} -D GEOGRAPHICLIB_LIB_TYPE=SHARED -D GEOGRAPHICLIB_DATA=${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}/share/data .
+      BUILD_COMMAND cd ${CMAKE_BINARY_DIR}/GeographicLib-prefix/src/GeographicLib && make -j1
+      INSTALL_COMMAND cd ${CMAKE_BINARY_DIR}/GeographicLib-prefix/src/GeographicLib && make install &&
+          ./tools/geographiclib-get-geoids -p ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}/share/data egm96-5 &&
+          ./tools/geographiclib-get-gravity -p ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}/share/data egm96 &&
+          ./tools/geographiclib-get-magnetic -p ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}/share/data emm2015
+  )
```

Comment:
This is horrible. I wouldn't want to install ~500MB of stuff just for this to work, and anyway most embedded targets on a drone may will not have internet connections. 

This change will greatly kill the portability and lightweight nature of mavros.